src/sage/schemes/projective/projective_space.py

```diff
@@ -2374,7 +2374,7 @@ def rational_points(self, bound=0):
             for ai in R:
                 P[i] = ai
                 for tup in S[i - 1]:
-                    if gcd([ai] + tup) == 1:
+                    if gcd([ai] + list(tup)) == 1:
```

Comment:
Here, `gcd((ai,) + tup)` is more appropriate.
```py
sage: def A(a, tup):
....:     return [a] + list(tup)
....: 
sage: def B(a, tup):
....:     return (a,) + tup
....: 
sage: tup = tuple(range(10))
sage: %timeit A(100, tup)
134 ns ± 0.0589 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
sage: %timeit B(100, tup)
84.8 ns ± 0.0729 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
sage: tup = tuple(range(4))
sage: %timeit A(100, tup)
123 ns ± 1.49 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
sage: %timeit B(100, tup)
79.1 ns ± 0.0314 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
sage: %timeit gcd(A(100, tup))
7.03 µs ± 4.32 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
sage: %timeit gcd(B(100, tup))
6.95 µs ± 11.7 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Reply:
That's not a fair comparison because of the extra (unnecessary) `list` cast, which creates a copy.

Reply:
> That's not a fair comparison because of the extra (unnecessary) `list` cast, which creates a copy. (Recall the old code was returning a list.)
I agree. Since the code now returns a tuple, it's better to use `(ai,) + tup` than casting to list.